berkdb/db/db_am.c

```diff
@@ -473,13 +502,35 @@ __db_count_cursors(DB *dbp)
 {
 	DBC *dbc;
 	int ncur = 0;
+	DB_CQ *cq;
+	DB_CQ_HASH *h;
+	void *hashent;
+	unsigned int pos;
+
+	if (!dbp->use_tlcq) {
+		MUTEX_THREAD_LOCK(dbp->dbenv, dbp->mutexp);
+		for (dbc = TAILQ_FIRST(&dbp->active_queue); dbc != NULL;
+			dbc = TAILQ_NEXT(dbc, links))
+			ncur++;
+		MUTEX_THREAD_UNLOCK(dbp->dbenv, dbp->mutexp);
+	} else {
+		Pthread_mutex_lock(&gbl_all_cursors.lk);
+		TAILQ_FOREACH(h, &gbl_all_cursors, links) {
+			Pthread_mutex_lock(&h->lk);
+			for (cq = hash_first(h->h, &hashent, &pos); cq != NULL;
```

Comment:
Does this hash use the db as the key?  Wondering if you could do a lookup by db here instead of traversing all elements ..

Reply:
Yes thanks for pointing out! I feel so dumb ... I changed it to `hash_find`.